daq/faucet_event_client.py

```diff
@@ -65,12 +69,25 @@ def has_event(self, blocking=False):
             if '\n' in self.buffer:
                 return True
             if blocking or self.has_data():
-                data = self.sock.recv(1024).decode('utf-8')
-                with self._buffer_lock:
-                    self.buffer += data
-            else:
+                data = self.sock.recv(2048).decode('utf-8')
+                if data.strip():
+                    self._last_message = time.time()
+                    with self._buffer_lock:
+                        self.buffer += data
+            if not blocking and self._check_socket_connection():
                 return False
 
+    def _check_socket_connection(self):
+        time_elapsed = time.time() - self._last_message
+        if self._event_timeout_sec and time_elapsed >= self._event_timeout_sec:
+            LOGGER.info('Attempting to reconnect to faucet socket %s ' %
+                        os.getenv('FAUCET_EVENT_SOCK'))
+            time.sleep(self._event_timeout_sec)  # Ensure faucet socket's been disconnected.
+            self.connect()
```

Comment:
as per the comments in runner.py, I think some kind of notification will be required for when the socket changes so that it can be properly monitored. Currently, it seems like the socket can change, creating a new fd, but it won't be properly monitored by the calling code -- how does this change even work as it is?